src/network/mod.rs

```diff
@@ -59,12 +62,35 @@ pub fn init(
 	// create swarm that manages peers and events
 	let swarm = {
 		let mut kad_cfg = KademliaConfig::default();
-		kad_cfg.set_query_timeout(Duration::from_secs(5 * 60));
+		kad_cfg
+			.set_publication_interval(Some(Duration::from_secs(
+				kad_config.publication_interval as u64,
+			)))
+			.set_replication_interval(Some(Duration::from_secs(
```

Comment:
Maybe we can make conversion during serde, so we can have valid duration here and avoid mixing conversions? Or at least to move it at `From` implementation for `KademliaConfig`?